lib/core/src/getRodsEnv.cpp

```diff
@@ -139,12 +140,11 @@ extern "C" {
         createRodsEnvDefaults( &rodsEnvArg );
     }
 
-    static
-    int capture_string_property(
-        const std::string&             _key,
-        char*                          _val ) {
+    static int capture_string_property(const std::string& _key, char* _val, size_t _val_size)
+    {
         try {
-            sprintf(_val, "%s", irods::get_environment_property<const std::string>(_key).c_str());
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-type-vararg)
+            snprintf(_val, _val_size, "%s", irods::get_environment_property<const std::string>(_key).c_str());
```

Comment:
Given that we're writing C++, please qualify C std functions with the `std::` namespace.

Make sure the correct header is included as well. I think it's `<cstring>`, but please verify.